src/main/java/Parser.java

```diff
@@ -0,0 +1,106 @@
+import java.text.ParseException;
+
+public class Parser {
+    private static int counter;
+
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param fullCommand the user input to parse.
+     * @throws DukeException for user input errors.
+     * @return Command to execute.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        Command c = null;
+        String[] commandArr = fullCommand.split(" ");
+        String[] keywords = {};
+        int i; // skip the task type
+        int taskId;
+
+        try {
+            switch (commandArr[0].toLowerCase()) {
```

Comment:
I like how you used the switch statements here, makes the code look much cleaner 👍